### PR TITLE
improve error message when module of package is not the same as the package

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

**Comment by vtjnash** [comment]
I have been considering how we might be able to remove this problem entirely

File: test/loading.jl [hunk 1]
```diff
@@ -745,3 +745,17 @@ end
         @test Base.get_deps(raw_manifest) == deps
     end
 end
+
+@testset "error message loading pkg bad module name"
```
Comment (DilumAluthge) [inline test/loading.jl:749]:
```suggestion
@testset begin "error message loading pkg bad module name"
```

Comment (KristofferC) [reply test/loading.jl]: Wait that was the wrong place for it :P

File: test/loading.jl [hunk 2]
```diff
@@ -745,3 +745,17 @@ end
...
 end
+
+@testset "error message loading pkg bad module name" begin
+    mktempdir() do tmp
+        old_loadpath = copy(LOAD_PATH)
+        try
+            push!(LOAD_PATH, tmp)
+            write(joinpath(tmp, "BadCase.jl"), "module badcase end")
+            @test_throws ErrorException("loading package `BadCase` did not define the expected module `BadCase`, \
+                                        check for typos in module name") (@eval using BadCase)
```
Comment (ararslan) [inline test/loading.jl:756]:
```suggestion
            @test_throws ErrorException("package `BadCase` did not define the expected module `BadCase`, \
                                        check for typos in package module name") (@eval using BadCase)
```

File: test/loading.jl (hunk 1 above)
Comment (DilumAluthge) [reply test/loading.jl]: 🤦

Comment (DilumAluthge) [reply test/loading.jl]: The blind leading the blind.

**Review by DilumAluthge — APPROVED** [review]

**Comment by DilumAluthge** [comment]
Any objections to merging this?

**Comment by KristofferC** [comment]
I think it is an improved user experience for now so I would say it could be merged.

**Pull request merged**